qcodes/instrument_drivers/Keysight/keysight_b220x.py

```diff
@@ -0,0 +1,181 @@
+                           )
+
+        self.add_function(name='gnd_disable_all_outputs',
+                          call_cmd=':AGND:CHAN:DIS:CARD 0')
+
+        self.add_function(name='gnd_disable_output',
+                          call_cmd=':AGND:CHAN:DIS (@001{:02d})',
+                          args=[Ints(1, 48), ])
+
+        self.add_function(name='gnd_enable_all_outputs',
+                          call_cmd=':AGND:CHAN:ENAB:CARD 0')
+
+        self.add_function(name='gnd_enable_output',
+                          call_cmd=':AGND:CHAN:ENAB (@001{:02d})',
+                          args=[Ints(1, 48), ])
+
+        self.add_parameter(name='gnd_input_port',
+                           get_cmd=':AGND:PORT? 0',
+                           set_cmd=':AGND:PORT 0,{}',
+                           vals=MultiType(Ints(1, 14),
+                                                     Enum(-1)),
+                           get_parser=lambda response: int(response),
+                           docstring="Selects the input that will be used as GND input port (default 12). The GND input port cannot be used on subsequent `connect` or `disconnect` commands if GND mode is ON"
+                           )
+
+        self.add_parameter(name='gnd_mode',
+                           get_cmd=':AGND? 0',
+                           set_cmd=':AGND 0,{}',
+                           val_mapping={True: 1,
+                                        False: 0}
+                           )
+
+        self.add_parameter(name='unused_inputs',
+                           get_cmd=':AGND:UNUSED? 0',
+                           set_cmd=":AGND:UNUSED 0,'{}'",
+                           get_parser=lambda response: [int(x) for x in response.strip("'").split(',') if x.strip().isdigit()],
+                           set_parser=lambda value: str(value).strip('[]'),
+                           vals=Lists(Ints(1, 14))
+                           )
+
+        self.add_parameter(name='couple_ports',
+                           get_cmd=':COUP:PORT? 0',
+                           set_cmd=":COUP:PORT 0,'{}'",
+                           set_parser=lambda value: str(value).strip('[]()'),
+                           get_parser=lambda response: [int(x) for x in response.strip("'").split(',') if x.strip().isdigit()],
+                           vals=Lists(Enum(1, 3, 5, 7, 9, 11, 13))
+                           )
+
+        self.add_function(name='couple_port_autodetect',
+                          call_cmd=':COUP:PORT:DET')
+
+        self.add_parameter(name='couple_mode',
+                           get_cmd=':COUP? 0',
+                           set_cmd=':COUP 0,{}',
+                           val_mapping={True: 1,
+                                        False: 0},
+                           docstring="Param: True for ON, False for OFF"
+                           )
+
+
+    def _connect(self, channel1, channel2):
+        self.write(":CLOS (@{card:01d}{ch1:02d}{ch2:02d})".format(card=0,
```

Comment:
please make the `card` a private attribute at least, so that "just in case" it can be easily set to whatever.

(i think you told me but i forgot) Also, is it possible for a single instrument to have multiple cards? if so, then i'd at least add a note in the driver's docstring that "multi-card operation" is not supported.

Reply:
I looked more carefully, and i see that the card number is in many commands. so, it's beneficial to put it an attribute because this will help when one will want to implement a driver for the multi-card case that is based on this driver.

Reply:
Good point. I will change that.

Reply:
Done. ✅ 
 - replaced the card parameter in all SCPI commands with a private variable self._card
 - added comment to driver doctoring that this driver only implements Auto mode.